xwiki-commons-core/xwiki-commons-cache/xwiki-commons-cache-infinispan/pom.xml

```diff
@@ -50,6 +50,11 @@
       <groupId>org.infinispan</groupId>
       <artifactId>infinispan-core</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.eclipse.microprofile.metrics</groupId>
+      <artifactId>microprofile-metrics-api</artifactId>
+      <version>3.0</version>
```

Comment:
I think I would put this version in a dependencyManagement along with the Infinispan one in the xwiki-commons root pom.

Reply:
Regarding commons, I don't remember why we put platform-related depMgmt there. At first sight it doesn't look legit and should be put in platform's pom.xml.

Reply:
>  I don't remember why we put platform-related depMgmt there

We don't.

Reply:
Then we shouldn't put the dep there then, since it's related to distribution wanting to bundle support for JMX by default.

Reply:
I don't understand what you mean. This is a dependency of xwiki-commons-cache-infinispan here, so I'm suggesting to move the version to xwiki-common pom.

If your goal was to suggest moving this dependency entirely to xwiki-platform then that's a different subject.

Reply:
I don't think it's a mandatory dep of xwiki-commons-cache-infinispan since we already a dep on infinispan core and it's not bringing it, which most likely means it's an optional dep. Right?

Reply:
It all depends if you want xwiki-commons-cache-infinispan to properly expose Infinispan JMX (which always been the case before) or not. Right now the regression is at xwiki-commons-cache-infinispan level.

Reply:
Yeah depends if we want to make the minimal flavor fatter or not I guess...

Reply:
> Yeah depends if we want to make the minimal flavor fatter or not I guess...

That dependency have to be in the WAR for Infinispan to find it anyway so it's not like putting it on platform side was going to change anything to the XWiki WAR size. Also we are talking about 34kB here.

Reply:
I was not referring to the XS WAR but to the minimal one which could be smaller with some optional features that not everyone might need.

Reply:
I'm +0 to add it to commons, not blocking it, just trying to think if it's the right thing to do.

Reply:
I moved the dependency declaration to xwiki-commons with the correct version and a comment. As for unit testing: I checked a bit how to proceed, not sure I'm able to provide a working one on my own for now though, any guidance or example welcome or I'll check how you did it.

Reply:
(moved the dependency in case it can be useful indeed despite not knowing either what's best)